test/e2e/common/pods.go

```diff
@@ -772,4 +772,83 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	It("pod should not be available after the graceful terimination delete call", func() {
+		pod := getProcessedPod(podClient, f)
+		By("waiting for pod get removed from the pods list")
+		Expect(wait.Poll(time.Second*5, time.Second*10, func() (bool, error) {
+			_, err := podClient.Get(pod.Name, metav1.GetOptions{})
+			if err == nil {
+				return false, err
+			}
+			return true, err
```

Comment:
I guess current non-existence way could cover the other exceptions accidentally.
How about using errors.IsNotFound() strictly like
```
return errors.IsNotFound(err), err
```